pallets/phala/src/registry.rs

```diff
@@ -373,6 +381,41 @@ pub mod pallet {
 			Ok(())
 		}
 
+		#[pallet::weight(0)]
+		pub fn update_worker_endpoint(
+			origin: OriginFor<T>,
+			signed_endpoint: SignedWorkerEndpoint,
+			signature: Vec<u8>,
+		) -> DispatchResult {
+			ensure_signed(origin)?;
+
+			// Validate the signature
+			ensure!(
+				signature.len() == 64,
+				Error::<T>::InvalidSignatureLength
+			);
+			let sig = sp_core::sr25519::Signature::try_from(signature.as_slice())
+				.or(Err(Error::<T>::MalformedSignature))?;
+			let encoded_data = signed_endpoint.versioned_endpoints.encode();
```

Comment:
You should verify the signature on the entire `signed_endpoint` object. Otherwise the other fields are not protected (i.e. the attacker can fake `signing_time`)

Reply:
Done